### PR TITLE
ffmpeg: add v6.1.1, drop old versions

### PR DESCRIPTION
**Pull request by valgur**
Also cleaned up package_info() a bit.

### CODE REVIEW

**Review by jcar87 — APPROVED** [review]

**Review by MartinDelille — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

File: recipes/ffmpeg/all/conanfile.py [hunk 1]
```diff
@@ -347,6 +349,10 @@ def validate(self):
                 raise ConanInvalidConfiguration("FFmpeg '{}' option requires '{}' option to be enabled".format(
                     dependency, "' or '".join(features)))
 
+        if conan_version.major == 1 and is_msvc(self) and self.options.shared:
+            # Linking fails with "Argument list too long" for some reason on Conan v1
+            raise ConanInvalidConfiguration("MSVC shared build is not supported for Conan v1")
```
Comment (SpaceIm) [inline recipes/ffmpeg/all/conanfile.py:354]: I'm confused. It works fine with previous versions but not for 6.1.1, but this PR now prevents shared msvc with conan v1 for these versions also?

File: recipes/ffmpeg/all/conanfile.py [hunk 2]
```diff
@@ -190,6 +190,8 @@ class FFMpegConan(ConanFile):
         "disable_filters": None,
         "enable_filters": None,
     }
+    # Fix for mslink: Argument list too long
+    short_paths = True
```
Comment (SpaceIm) [inline recipes/ffmpeg/all/conanfile.py:194]: Did it fix anything in the end?

Comment (valgur) [reply recipes/ffmpeg/all/conanfile.py:194]: No, it didn't.

File: recipes/ffmpeg/all/conanfile.py (hunk 1 above)
Comment (valgur) [reply recipes/ffmpeg/all/conanfile.py:354]: It actually fails for all versions - I tested with disabling it just for v6.1.1, but it did not help and I force-pushed over it (sorry about that): https://github.com/conan-io/conan-center-index/compare/a4db9e764789ae8bca3f91b24d7e49e5dae2a276..395f7f6c00511abb9a88d2182d2c2e18959cef1c

Comment (SpaceIm) [reply recipes/ffmpeg/all/conanfile.py:354]: shared was building fine in this pass: https://github.com/conan-io/conan-center-index/pull/22195#issuecomment-2119346289

**Pull request merged**